layouts/partials/single_article.html

```diff
@@ -19,7 +19,7 @@
         {{ partial "prev_next_post" . }}
     </article>
 
-    {{ if not .Params.disable_comments }}
+    {{ if and (not (eq .Site.DisqusShortname "")) (not .Params.disable_comments) }}
```

Comment:
The disqus shortcode already [checks](https://github.com/spf13/hugo/blob/2c5e4f7640e71d2a193a74e6c41109ec40bc0222/tpl/template_embedded.go#L131) if `.Site.DisqusShortname` is an empty string.